vulture_os/toolkit/api_parser/perception_point_x_ray/perception_point_x_ray.py

```diff
@@ -0,0 +1,226 @@
+                        f"Wait a minute before continue. (retry {retry})",
+                        extra={'frontend': str(self.frontend)})
+                    retry += 1
+                    self.evt_stop.wait(60.0)
+                    continue
+
+                else:
+                    logger.error(f"[{__parser__}]:__execute_query: Error at PerceptionPointXRay API Call URL: {url} Code: {response.status_code} Content: {response.content}\n" \
+                        f"Response : {response}", extra={'frontend': str(self.frontend)})
+                    retry += 1
+                    self.evt_stop.wait(10.0)
+                    continue
+
+        msg = f"Error at PerceptionPointXRay API Call URL: {url}: Unable to get a correct response"
+        logger.error(f"[{__parser__}]:__execute_query: {msg}", extra={'frontend': str(self.frontend)})
+        raise PerceptionPointXRayAPIError(msg)
+
+
+
+    def _connect(self):
+        if self.session is None:
+            self.session = Session()
+            self.session.headers.update(self.HEADERS)
+            self.session.headers.update({"Authorization": f"Token {self.token}"})
+            self.session.headers.update({"User-Agent": ""})
+        return True
+
+    def test(self):
+        try:
+            if self.session is None:
+                self._connect()
+
+            since = timezone.now() - timedelta(days=30)
+            to = timezone.now()
+
+            response, _, _, _ = self.get_logs(since, to, self.LOG_TYPES[0], None)
+
+            return {
+                "status": True,
+                "data": response
+            }
+        except Exception as e:
+            logger.exception(f"[{__parser__}]:test: {e}", extra={'frontend': str(self.frontend)})
+            return {
+                "status": False,
+                "error": str(e)
+            }
+
+    def get_scan_details(self, scan_id):
+        # Get details for every scan
+        url = f"{self.perception_point_x_ray_host}/api/v1/scans/list/{scan_id}/"
+        return self._execute_query(url)
+
+    def get_last_scans(self, since, to, log_type, next=None):
+        if self.session is None:
+            self._connect()
+
+        if next:
+            url = next
+            payload = {}
+        else:
+            url = f"{self.perception_point_x_ray_host}/api/v1/scans/list/"
+            payload = {
+                "end": int(to.timestamp()),
+                "start": int(since.timestamp()),
+                "verbose_verdict": log_type,
+            }
+
+        return self._execute_query(url, payload)
+
+    def get_logs(self, since, to, log_type, next=None):
+        last_scans = {}
+        while not self.evt_stop.is_set():  # Get the ideal timerange of logs to avoid errors if API returns more of 1000 logs
+            last_scans = self.get_last_scans(since, to, log_type, next)
+            if last_scans.get('count') >= 1000:
```

Comment:
do you mean the API cannot/doesn't return more than 1000 entries correctly?
comment is unclear

Reply:
API returns an error if we fetch more than 1000 logs at a time

Reply:
it the error raised by the API called by `get_last_scans()` ?